source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -527,6 +527,48 @@ void Filter::sendTrailers(ProcessorState& state, const Http::HeaderMap& trailers
   stats_.stream_msgs_sent_.inc();
 }
 
+void Filter::onNewTimeout(
+    const std::unique_ptr<envoy::service::ext_proc::v3::ProcessingResponse>&& response) {
+  // In the following conditions the function just bails out and the message is ignored.
+  // 1) The message_timeout field is not set, or even it is set but < 1ms;
+  // 2) or the traffic_direction field is not set to be INBOUND or OUTBOUND.
+  const auto new_timeout = response->new_timeout();
+  const auto message_timeout_ms_opt = PROTOBUF_GET_OPTIONAL_MS(new_timeout, message_timeout);
+  if (!message_timeout_ms_opt) {
+    ENVOY_LOG(warn, "Ext_proc server new timeout setting is missing. Ignoring the message.");
+    return;
+  }
+  const auto message_timeout_ms = message_timeout_ms_opt->count();
+  const uint32_t min_timeout_ms = 1;
+  const uint32_t max_timeout_ms = 10000;
+  if (message_timeout_ms < min_timeout_ms || message_timeout_ms > max_timeout_ms) {
+    ENVOY_LOG(warn, "Ext_proc server new timeout setting is out of range. "
+                    "Ignoring the message.");
+    return;
+  }
+  const auto traffic_direction = new_timeout.traffic_direction();
```

Comment:
const auto&

Reply:
Regarding the traffic_direction, I decided to not let the server send it down. Instead, let the envoy ext_proc filter to figure out which traffic direction it is processing, i.e, which timer() to stop/restart.  This can be achieved since at any given time eith downstream request or upstream response is being processed, and only one timer is running.